internal/provider/resource/project_role_resource.go

```diff
@@ -92,9 +116,41 @@ func (r *projectRoleResource) Schema(_ context.Context, _ resource.SchemaRequest
 				Computed:      true,
 				PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
 			},
+			"permissions_v2": schema.SetNestedAttribute{
+				Optional:    true,
+				Description: "The permissions assigned to the project role. Refer to the documentation here https://infisical.com/docs/internals/permissions for its usage.",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"action": schema.SetAttribute{
+							ElementType: types.StringType,
+							Description: "Describe what actions an entity can take.",
+							Required:    true,
+						},
+						"subject": schema.StringAttribute{
+							Description: "Describe the entity the permission pertains to.",
+							Required:    true,
+						},
+						"inverted": schema.BoolAttribute{
+							Description: "Whether rule forbids. Set this to true if permission forbids.",
+							Required:    true,
+						},
+						"conditions": schema.StringAttribute{
+							Optional:    true,
+							Description: "When specified, only matching conditions will be allowed to access given resource.",
+							PlanModifiers: []planmodifier.String{
+								pkg.JsonEquivalentModifier{},
+							},
+							Validators: []validator.String{
+								infisicaltf.JsonStringValidator,
+							},
+						},
+					},
+				},
+			},
 			"permissions": schema.ListNestedAttribute{
-				Required:    true,
-				Description: "The permissions assigned to the project role",
+				Optional:           true,
+				DeprecationMessage: "Use permissions_v2 instead as it allows you to be more granular with access control",
+				Description:        "(DEPRECATED, USE permissions_v2) The permissions assigned to the project role",
```

Comment:
I think it will be really helpful to point them to a document where we tell them how they can move from permisions v1 to v2 in terraform. The reason is because you can't just paste in the v1 permissions into v2. So i think people will be like how do i make this work. also since during each apply they get a big warning, i think they would want to know how they can fix this properly. what do you think?

Reply:
agreed... added reference to the migration guide. 

It's part of the PR here:
https://github.com/Infisical/infisical/pull/2689